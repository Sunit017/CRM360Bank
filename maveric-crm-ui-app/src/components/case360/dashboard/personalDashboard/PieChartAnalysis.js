import React from "react";
import { Box, Grid } from "@mui/material";
import styled from "styled-components";
import PieChart from "./PersonalDashboardPieChart";

const PieChartAnalysis = ({ category }) => {
  const statusLabels = ["New", "Open", "Closed", "Resolved", "Reopened"];
  const agedCasesAndResolvedLabels = [
    "3 Days and Below",
    "4 to 7 Days",
    "8 to 13 Days",
    "14 Days and Above",
  ];

  const complaintsStatusSeries = [30, 40, 45, 50, 49];
  const complaintsAgedCasesSeries = [30, 40, 45, 50];
  const complaintsResolvedSeries = [30, 40, 45, 50];

  return (
    <Box sx={{ maxWidth: "100%" }}>
      <Grid
        container
        spacing={{ xs: 2, md: 3, lg: 1 }}
        columns={{ xs: 4, sm: 8, md: 12 }}
      >
        <Grid item xs={4}>
          <Box sx={{ border: "1px solid black", minHeight: "250px" }}>
            <PieChart
              headerTitle={`${category} - Status`}
              type="pie"
              casesLabel={statusLabels}
              data={complaintsStatusSeries}
            />
          </Box>
        </Grid>
        <Grid item xs={4}>
          <Box sx={{ border: "1px solid black", minHeight: "250px" }}>
            <PieChart
              headerTitle={`${category} - Aged Cases`}
              type="pie"
              casesLabel={agedCasesAndResolvedLabels}
              data={complaintsAgedCasesSeries}
            />
          </Box>
        </Grid>
        <Grid item xs={4}>
          <Box sx={{ border: "1px solid black", minHeight: "250px" }}>
            <PieChart
              headerTitle={`${category} - Resolved`}
              type="pie"
              casesLabel={agedCasesAndResolvedLabels}
              data={complaintsResolvedSeries}
            />
          </Box>
        </Grid>
      </Grid>
    </Box>
  );
};

export default PieChartAnalysis;
