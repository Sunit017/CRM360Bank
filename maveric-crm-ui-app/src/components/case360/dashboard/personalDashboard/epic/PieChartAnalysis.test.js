import renderWithRedux from "../../../../../testUtil";
import { cleanup } from "@testing-library/react";
import PieChartAnalysis from "../PieChartAnalysis";

const Chart = () => null;
jest.mock("react-apexcharts", () => Chart);

afterEach(cleanup);

test("Render pie charts", () => {
  const container = renderWithRedux(<PieChartAnalysis category={"Test"} />, {});
  expect(container).toMatchSnapshot();
});
