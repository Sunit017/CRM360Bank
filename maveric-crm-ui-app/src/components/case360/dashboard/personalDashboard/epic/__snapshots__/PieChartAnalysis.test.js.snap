// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render pie charts 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-9whsf3"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 MuiGrid-spacing-lg-1 css-vobqoz-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-iltpxr-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-huleca"
            >
              <div
                style="width: 50%; margin: 5px 5px 5px 0px;"
              >
                <div
                  style="background-color: rgb(9, 47, 133); color: rgb(255, 255, 255); font-size: small; height: 25px; padding-left: 10px; padding-top: 5px; width: 100%;"
                >
                  Test - Status
                </div>
                <div />
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-iltpxr-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-huleca"
            >
              <div
                style="width: 50%; margin: 5px 5px 5px 0px;"
              >
                <div
                  style="background-color: rgb(9, 47, 133); color: rgb(255, 255, 255); font-size: small; height: 25px; padding-left: 10px; padding-top: 5px; width: 100%;"
                >
                  Test - Aged Cases
                </div>
                <div />
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-iltpxr-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-huleca"
            >
              <div
                style="width: 50%; margin: 5px 5px 5px 0px;"
              >
                <div
                  style="background-color: rgb(9, 47, 133); color: rgb(255, 255, 255); font-size: small; height: 25px; padding-left: 10px; padding-top: 5px; width: 100%;"
                >
                  Test - Resolved
                </div>
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-9whsf3"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 MuiGrid-spacing-lg-1 css-vobqoz-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-iltpxr-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-huleca"
          >
            <div
              style="width: 50%; margin: 5px 5px 5px 0px;"
            >
              <div
                style="background-color: rgb(9, 47, 133); color: rgb(255, 255, 255); font-size: small; height: 25px; padding-left: 10px; padding-top: 5px; width: 100%;"
              >
                Test - Status
              </div>
              <div />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-iltpxr-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-huleca"
          >
            <div
              style="width: 50%; margin: 5px 5px 5px 0px;"
            >
              <div
                style="background-color: rgb(9, 47, 133); color: rgb(255, 255, 255); font-size: small; height: 25px; padding-left: 10px; padding-top: 5px; width: 100%;"
              >
                Test - Aged Cases
              </div>
              <div />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-iltpxr-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-huleca"
          >
            <div
              style="width: 50%; margin: 5px 5px 5px 0px;"
            >
              <div
                style="background-color: rgb(9, 47, 133); color: rgb(255, 255, 255); font-size: small; height: 25px; padding-left: 10px; padding-top: 5px; width: 100%;"
              >
                Test - Resolved
              </div>
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
